cmake/Platform/Common/Install_common.cmake

```diff
@@ -243,6 +235,32 @@ set_property(TARGET ${TARGET_NAME}
 endfunction()
 
 
+#! ly_platform_install_target: platform specific target installation
+function(ly_platform_install_target)
```

Comment:
this should also receive `archive_output_directory`, `library_output_directory` and `runtime_output_directory` 
the function is kind of taking some parameters one way and these other parameters another way

Reply:
this function is also kind of abusing the override mechanism in functions. We prefer to put the function in another function that Linux/Windows include whereas Mac implements its own (similar to PAL)

Reply:
`library_output_directory` and `runtime_output_directory ` are set at the top of the file. They're not being passed in as parameters to this function as well. It looks like `runtime_output_directory` is being used in multiple functions in the Install_common file. Should I move `archive_output_directory` to the top of the file as well? Otherwise, we need to set `runtime_output_directory` every function it's being used.

For the Linux/Windows implementation, I agree. The reason I went with this is because we only need this hack for Mac. Eventually, I think Mac can use the common implementation as well. We can move the BUNDLE/RESOURCE DESTINATION to the common version as they'll be ignored on platforms that don't support them. We can have just one implementation of the Install at that point once this hack is no longer necessary.

Reply:
I see... yeah, I would pass them as parameters anyways, that way the function is more "encapsulated" to where it is.

For the function override, is ok if you go this way, but leave a comment in the override so someone is not tempted to move that function around.

Reply:
I think we should just call the platform specific version of `ly_platform_install_target` a different name.

CMake supports  an [if(COMMAND)](https://cmake.org/cmake/help/v3.21/command/if.html#existence-checks) check checking whether a function can be invoked

This function could do something such as
```
if(COMMAND ly_platform_install_target_impl)
    ly_platform_install_target_impl(${ARGN})
else()
    # The rest of the common function implementation
endif()
```

Reply:
I'll try this out. I locked myself out of my Mac. I'll update tomorrow.

Reply:
some internal pieces of CMake call this function "override", e.g. `BundleUtilities.cmake`
I am ok with `_impl`, `_override`  seems would be a better match of the intention

I also feel the `platform` in the name is a bit confusing since the implementation is not "per platform" but just "overridden by one platform", what about `ly_install_target_override` (and the common function `ly_install_target`)

Reply:
I am good with `ly_install_target_override` as the name of the function.

Reply:
This has been addressed.